core/src/main/scala/akka/persistence/r2dbc/session/scaladsl/R2dbcSession.scala

```diff
@@ -29,12 +28,15 @@ object R2dbcSession {
    * transaction is committed at the end or rolled back in case of failures.
    */
   def withSession[A](system: ActorSystem[_])(fun: R2dbcSession => Future[A]): Future[A] = {
-    withSession(system, "akka.persistence.r2dbc.connection-factory")(fun)
+    // FIXME parsing these for each query is no good
+    val settings = R2dbcSettings(system.settings.config.getConfig("akka.persistence.r2dbc"))
+    withSession(system, settings, s"akka.persistence.r2dbc.${settings.dialectName}.connection-factory")(fun)
   }
 
-  def withSession[A](system: ActorSystem[_], connectionFactoryConfigPath: String)(
+  def withSession[A](system: ActorSystem[_], settings: R2dbcSettings, connectionFactoryConfigPath: String)(
```

Comment:
see my comment in ConnectionFactoryProvider, but if we have to add this variant we should still keep the old signature even though it is ApiMayChange

Reply:
Being able to keep the old signature now 👍